### DISCUSSION TITLE
about tlb lookup latency

### DISCUSSION THREAD

**Discussion by InnoBogus** [sarchlab/mgpusim#108] [Q&A] [answered]
Hello author, I want to consult about tlb related questions. I found that other components or operations are set with fixed delay, but tlb is not. When I add fixed delay to tlb, kernal time is greatly reduced. I want to know whether the search delay of tlb really exists in this simulator?

**Comment by syifan** [comment]
@DX990307 Would you be able to answer this question?

**Comment by DX990307** [comment]
Hi, I am trying to answer your question. Sure, the TLB component definitely has the search delay in our simulator. Since the TLB cannot hold all page addresses. When encountering TLB-miss, the TLB will request lower components, such as L2 TLB or IOMMU, for further search. This will consume a lot of time. If you set a fixed time delay and this delay is not very long, the kernel time would be greatly reduced.

**Reply by InnoBogus** [reply]
Thank you very much！Could you please tell me how tlb lookup delay is defined or calculated? Because I see that other components, such as cache or noc connections have clear latency definitions, such as 100cycle, but I don't see them in tlb.

**Reply by DX990307** [reply, marked as answer]
Got it, I am sorry that I misunderstood your question. What I mean that the TLB have search latency, but this latency is caused by TLB-miss. However, it does not set a certain "search delay"  like 100 cycles in TLB itself.
